### ISSUE TITLE
Additional information for gateway/v1.AddressType

### ISSUE THREAD

**Issue by cccsss01** [kubernetes-sigs/gateway-api#3476]
**What would you like to be added**:
For hostname, is wild card supported?
Add a dns example to gateway address

**Why this is needed**:
https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.AddressType
https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GatewayAddress

**Comment by youngnick** [comment]
No, wild card is not supported, because the intent of being able to set a hostname for Addresses is to support use cases like AWS load balancers, where the IP addresses rotate, and so the best way to connect to the listener is via a DNS name rather than an IP address.

In all cases, the Address is used as the place for clients to connect to, so having a wildcard DNS entry doesn't make any sense.
